tests/test_meshtastic_utils_health.py

```
@@ -64,21 +85,29 @@ async def test_check_connection_ble_skips_health_checks(reset_meshtastic_globals
+    loop.run_in_executor.assert_called_once()
+    _, probe = loop.run_in_executor.call_args.args
+    assert probe.func is mu._get_device_metadata
+    assert probe.args == (mu.meshtastic_client,)
+    assert probe.keywords == {"force_refresh": True, "raise_on_error": True}
+    mock_logger.error.assert_not_called()
 
 
 @pytest.mark.asyncio
-async def test_check_connection_metadata_fallback_succeeds(reset_meshtastic_globals):
+async def test_check_connection_triggers_reconnect_on_probe_failure(
+    reset_meshtastic_globals,
```

Ruff (0.15.4)

[warning] Unused function argument: `reset_meshtastic_globals`

(ARG001)